java-frontend/src/test/java/org/sonar/java/ast/parser/grammar/statements/SwitchStatementTest.java

```diff
@@ -165,27 +155,23 @@ public void variable_declaration_in_a_case() {
         "switch (i) {" +
         "  case 1 -> { char a = 'A'; print(a); }" +
         "}")
-      // simplified control flow does not support variable declaration without block
-      .notMatches("" +
+      // Our parser is more tolerant than java 12, we wrongly accept variable declaration without block
+      .matches("" +
         "switch (i) {" +
         "  case 1 -> char a = 'A';" +
         "}");
   }
 
-  // TODO to support java 12
-  @Ignore
   @Test
   public void switch_should_be_a_statement_with_or_without_a_semicolon() {
     assertThat(JavaLexer.STATEMENT)
-      .matches("switch (i) { case 1 -> print('A'); }")
-      .matches("switch (i) { case 1 -> print('A'); };")
+      .matches("{ switch (i) { case 1 -> print('A'); } }")
```

Comment:
this is NOT testing that it's a statement anymore, now for the next two lines we endup testing that the block is a statement. This is not was the test was doing.